ui/decredmaterial/editor.go

```diff
@@ -62,3 +114,95 @@ func (e Editor) Layout(gtx *layout.Context, editor *widget.Editor) {
 	editor.PaintCaret(gtx)
 	stack.Pop()
 }
+
+func (e EditorCustom) Layout(gtx *layout.Context) {
+	e.handleEvents(gtx)
+	if e.IsVisibleBtn {
+		e.flexWidth = 0.93
+	}
+
+	layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+		layout.Rigid(func() {
+			if e.editor.Text() == "" {
+				e.theme.Body1("").Layout(gtx)
+			} else {
+				e.theme.Body1(e.title).Layout(gtx)
+			}
+		}),
+		layout.Rigid(func() {
+			layout.Flex{}.Layout(gtx,
+				layout.Rigid(func() {
+					layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+						layout.Rigid(func() {
+							inset := layout.Inset{
+								Top:    unit.Dp(6),
+								Bottom: unit.Dp(4),
+							}
+							inset.Layout(gtx, func() {
+								layout.Flex{}.Layout(gtx,
+									layout.Flexed(e.flexWidth, func() {
+										e.editorMaterial.Layout(gtx, e.editor)
+									}),
+								)
+							})
+						}),
+						layout.Rigid(func() {
+							layout.Flex{}.Layout(gtx,
+								layout.Flexed(e.flexWidth, func() {
+									rect := f32.Rectangle{
+										Max: f32.Point{
+											X: float32(gtx.Constraints.Width.Max),
+											Y: 1,
+										},
+									}
+									op.TransformOp{}.Offset(f32.Point{
+										X: 0,
+										Y: 0,
+									}).Add(gtx.Ops)
+									paint.ColorOp{Color: e.LineColor}.Add(gtx.Ops)
+									paint.PaintOp{Rect: rect}.Add(gtx.Ops)
+								}),
+							)
+						}),
+					)
+				}),
+				layout.Rigid(func() {
+					inset := layout.Inset{
+						Left: unit.Dp(10),
+					}
+					inset.Layout(gtx, func() {
+						if e.IsVisibleBtn {
+							if e.editor.Text() == "" {
+								e.pasteBtnMaterial.Layout(gtx, e.pasteBtnWidget)
+							} else {
+								e.clearBtMaterial.Layout(gtx, e.clearBtnWidget)
+							}
+						}
+					})
+				}),
+			)
+		}),
+	)
+}
+
+func (e EditorCustom) handleEvents(gtx *layout.Context) {
+	data, err := clipboard.ReadAll()
+	if err != nil {
+		panic(err)
+	}
+
+	for e.pasteBtnWidget.Clicked(gtx) {
+		e.editor.SetText(data)
+	}
+	for e.clearBtnWidget.Clicked(gtx) {
+		e.editor.SetText("")
+	}
+
+	for _, evt := range e.editor.Events(gtx) {
+		switch evt.(type) {
+		case widget.ChangeEvent:
+			e.editorMaterial.HintColor = e.theme.Color.Hint
```

Comment:
Why is this necessary?

Reply:
Since we have codes that had to use the events on the editor on handle_input.go, i felt its easier we eliminate those repeating codes by adding it here instead.

Reply:
But the editor's `HintColor` can't change since its not exported.